### ISSUE TITLE
Shader updates conflict with game updates ? Launch workflow disrupted

### ISSUE THREAD

**Issue by Saroumane** [ValveSoftware/steam-for-linux#7885]
#### Your system information

Steam client version (build number or date): Jun 8 2021
Distribution (e.g. Ubuntu): Ubuntu 21.04
Opted into Steam client beta?: No
Have you checked for system updates?: Yes

#### What happens 

Step 1
Launch Dota2, check "Launch game as soon as it's ready"
Steam client downloads Dota2 shaders
Small inconsistency : "Ready to launch" is shown
![image](https://user-images.githubusercontent.com/22261137/123513564-2470bb80-d68e-11eb-9062-0e93dfc2ec0d.png)

Step 2
Launch fails with the pop up below
![image](https://user-images.githubusercontent.com/22261137/123513578-318daa80-d68e-11eb-8c66-8c0d838987a0.png)

In the background, Steam client downloads a game update.
Wait the game update to finish.
Press OK to dismiss the pop up.

Step 3
Launch Dota2.
A pop up shows "vulkan processing", then the game starts

Note : it's not game specific at all, I just reproduced this with Tekken 7.

#### What should happen 

Launch Dota2, check "Launch game as soon as it's ready"
All updates are handled automatically, then the game starts without further user interaction.

**Comment by Turmfalke2** [comment]
Yes, every time there is 5mb localization update for dota 2 steam downloads 500mb shader cache. Isn't the shader cache supposed to be calculated client side? Also why would every update invalidate the complete shader cache?
